Lib/test/test_asyncio/test_base_events.py

```diff
@@ -1044,6 +1044,27 @@ def test_asyncgen_finalization_by_gc_in_other_thread(self):
             test_utils.run_briefly(self.loop)
             self.assertTrue(status['finalized'])
 
+    def test_shutdown_asyncgens(self):
+        # gh-117536: Test shutdown_asyncgens() using asyncio.run() which
+        # may cancel the task which closes the asynchronous generator before
+        # calling shutdown_asyncgens().
+
+        ns = {'state': 'not started'}
+        async def agen():
+            try:
+                ns['state'] = 'started'
+                yield 0
+                yield 1
+            finally:
+                ns['state'] = 'finalized'
```

Comment:
This should await something to prove it's getting aclosed properly

```suggestion
                await asyncio.sleep(0)
                ns['state'] = 'finalized'
```

Reply:
Right. I added a second test. I prefer to have two tests, since having 'await' also changes the behavior in a subtle way.